### PR TITLE
fix: Incorrect fields in the docs for access_code

### LINKED ISSUE #4771: Inconsistent Model and API docs for Access code.
**Describe the bug**
The defined model and the API docs differ for Access code.

##### API docs
Parameter | Description | Type | Required
-- | -- | -- | --
code | Access Code | string | -
access-url | URL of the access code | string | -
type | Amount or Percentage | string | -
is-active | If the access code is active | boolean | -
tickets-number | Tickets Number for Access Code | integer (>=0) | -
min-quantity | Minimum number of tickets for Access Code | integer (>=0) | -
max-quantity | Maximum number of tickets for Access Code | integer (>=0) | -
valid-from | Access Code valid from | ISO 8601 (tz-aware) | -
valid-till | Access Code valid till | ISO 8601 (tz-aware) | -
created-at | Access Code created at | ISO 8601 (tz-aware) | -
tickets | Event or Ticket | String | -
used-for | Access Code used for | String | -

##### Server
``` Python
    id = db.Column(db.Integer, primary_key=True)
    code = db.Column(db.String)
    access_url = db.Column(db.String)
    is_active = db.Column(db.Boolean)
    tickets_number = db.Column(db.Integer)  # For event level access this holds the max. uses
    min_quantity = db.Column(db.Integer)
    max_quantity = db.Column(db.Integer)  # For event level access this holds the months for which it is valid
    valid_from = db.Column(db.DateTime(timezone=True), nullable=True)
    valid_till = db.Column(db.DateTime(timezone=True), nullable=True)
    event_id = db.Column(db.Integer, db.ForeignKey('events.id', ondelete='CASCADE'))
    event = db.relationship('Event', backref='access_codes', foreign_keys=[event_id])
    created_at = db.Column(db.DateTime(timezone=True))
    marketer_id = db.Column(db.Integer, db.ForeignKey('users.id', ondelete='CASCADE'))
    marketer = db.relationship('User', backref='access_codes_')

    used_for = db.Column(db.String)
```
1. ```type```: Mentioned in the API docs but absent in the model.
2. ```tickets``` is also absent in the model. Moreover it's use is not very clear.

**Expected behavior**
There shouldn't be differences.

### PR DESCRIPTION
**Pull request by srv-twry**
Fixes #4771 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Removes the incorrect fields which are not present in the model but mentioned in the docs. 

#### Changes proposed in this pull request:

- Removes field ```type``` and ```ticket``` from the docs as type is not in the model while ticket is a many to many relationship.
- Made ```code``` a required field and added in the docs
- Added ```use_defaults``` decorator to add default values before schema validation.

### CODE REVIEW

**Comment by srv-twry** [comment]
@bhaveshAn @mayank8318 @schedutron Please review :)

File: docs/api/api_blueprint.apib [hunk 1]
```diff
@@ -16235,17 +16235,15 @@ Access Codes related to the events.
...
-| `type` | Amount or Percentage | string | - |
 | `is-active` | If the access code is active | boolean | - |
 | `tickets-number` | Tickets Number for Access Code | integer (>=0) | - |
 | `min-quantity` | Minimum number of tickets for Access Code | integer (>=0) | - |
 | `max-quantity` | Maximum number of tickets for Access Code | integer (>=0) | - |
 | `valid-from` | Access Code valid from | ISO 8601 (tz-aware) | - |
 | `valid-till` | Access Code valid till | ISO 8601 (tz-aware) | - |
 | `created-at` | Access Code created at | ISO 8601 (tz-aware) | - |
-| `tickets` | Event or Ticket | String | - |
 | `used-for` | Access Code used for | String | - |
```
Comment (srv-twry) [inline docs/api/api_blueprint.apib]: @niranjan94 One query. Is the behaviour of access code supposed to be similar to discount code ? Will there be two types of access codes i.e. one created by the super_admin and used for all the events and second created by the event organiser and only used inside that event ?

Comment (niranjan94) [reply docs/api/api_blueprint.apib]: No. There is only one type of access code. `created by the event organiser and only used inside that event`

Comment (srv-twry) [reply docs/api/api_blueprint.apib]: Alright, then we don't need the variable ```used-for``` as well.

**Pull request merged**